src/components/implementation/no_interface/cFE_booter/osfiles.c

```diff
@@ -228,33 +223,44 @@ int32 OS_rmdir(const char *path)
 int32 OS_mkfs(char *address,char *devname, char *volname,
                                 uint32 blocksize, uint32 numblocks)
 {
-    PANIC("Unimplemented method!"); // TODO: Implement me!
-    return 0;
+    uint32 ret = 0;
+    if (address) return OS_FS_ERR_INVALID_POINTER;
+    if (!devname || !volname) return OS_FS_ERR_INVALID_POINTER;
+    if (strlen(devname) >= OS_FS_DEV_NAME_LEN || strlen(volname) >= OS_FS_VOL_NAME_LEN)
+        return OS_FS_ERROR;
+    ret = newfs_init(devname, volname, blocksize, numblocks);
+    if (ret) return ret;
+    ret = tar_parse();
+    if (ret) return ret;
```

Comment:
You should explicitly compare against OS_SUCCESS. You shouldn't assume it's always 0.